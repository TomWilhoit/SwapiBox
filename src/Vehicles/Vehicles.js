import React, { Component } from "react";
import "../Vehicles/Vehicles.scss";

class Vehicles extends Component {
  constructor() {
    super();
  }

  returnVehicles = () => {
    let vehicleItem = this.props.vehiclesArray.map((vehicle, index) => {
      return (
        <div key={index} className="card">
          <p>Name: {vehicle.name}</p>
          <p>Model: {vehicle.model}</p>
          <p>Class: {vehicle.vehicle_class}</p>
          <p>Number of Passengers:{vehicle.passengers}</p>
        </div>
      );
    });
    return vehicleItem;
  };

  render() {
    if (this.props.vehiclesArray.length === 0) {
      return <div className="loading">Loading</div>;
    } else {
      return (
        <div className="Vehicles">
          <h1 className="title">Vehicles</h1>
          <div className="button-container">
            <button className="btn" onClick={this.props.goToMain}>
              MAIN
            </button>
            <button className="btn" onClick={this.props.goToPeople}>
              PEOPLE
            </button>
            <button className="btn" onClick={this.props.goToPlanets}>
              PLANETS
            </button>
          </div>
          <div className="card-container">{this.returnVehicles()}</div>
        </div>
      );
    }
  }
}

export default Vehicles;
