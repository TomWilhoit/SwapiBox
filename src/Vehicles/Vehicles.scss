body{
  background-color: lightblue;
}

.button-container {
  display: flex;
  justify-content: space-around
}

.card{
  border-style: solid;
  background-color: white;
  width: 200px;
  margin: 25px auto;
  border-radius: 10px
}

.card-container {
  display: grid;
  grid-template-columns: 25% 25% 25% 25%
}

.title {
  text-align: center
}