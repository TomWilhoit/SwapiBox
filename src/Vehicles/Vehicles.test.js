import React from "react";
import Vehicles from "../Vehicles/Vehicles";
import { shallow } from "enzyme";

describe("People", () => {
  let wrapper;
  let vehiclesArray = [1,2,3]

  beforeEach(() => {
    wrapper = shallow(<Vehicles vehiclesArray={vehiclesArray} />);
  });

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
