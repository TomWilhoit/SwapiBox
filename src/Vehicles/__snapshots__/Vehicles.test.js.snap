// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People should match snapshot when all data is passed correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    vehiclesArray={
      Array [
        1,
        2,
        3,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Vehicles
        </h1>,
        <div
          className="button-container"
        >
          <button
            className="btn"
          >
            MAIN
          </button>
          <button
            className="btn"
          >
            PEOPLE
          </button>
          <button
            className="btn"
          >
            PLANETS
          </button>
        </div>,
        <div
          className="card-container"
        >
          <div
            className="card"
          >
            <p>
              Name: 
            </p>
            <p>
              Model: 
            </p>
            <p>
              Class: 
            </p>
            <p>
              Number of Passengers:
            </p>
          </div>
          <div
            className="card"
          >
            <p>
              Name: 
            </p>
            <p>
              Model: 
            </p>
            <p>
              Class: 
            </p>
            <p>
              Number of Passengers:
            </p>
          </div>
          <div
            className="card"
          >
            <p>
              Name: 
            </p>
            <p>
              Model: 
            </p>
            <p>
              Class: 
            </p>
            <p>
              Number of Passengers:
            </p>
          </div>
        </div>,
      ],
      "className": "Vehicles",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": "title",
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn"
            >
              MAIN
            </button>,
            <button
              className="btn"
            >
              PEOPLE
            </button>,
            <button
              className="btn"
            >
              PLANETS
            </button>,
          ],
          "className": "button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MAIN",
              "className": "btn",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "MAIN",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PEOPLE",
              "className": "btn",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "PEOPLE",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PLANETS",
              "className": "btn",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "PLANETS",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card"
            >
              <p>
                Name: 
              </p>
              <p>
                Model: 
              </p>
              <p>
                Class: 
              </p>
              <p>
                Number of Passengers:
              </p>
            </div>,
            <div
              className="card"
            >
              <p>
                Name: 
              </p>
              <p>
                Model: 
              </p>
              <p>
                Class: 
              </p>
              <p>
                Number of Passengers:
              </p>
            </div>,
            <div
              className="card"
            >
              <p>
                Name: 
              </p>
              <p>
                Model: 
              </p>
              <p>
                Class: 
              </p>
              <p>
                Number of Passengers:
              </p>
            </div>,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Name: 
                </p>,
                <p>
                  Model: 
                </p>,
                <p>
                  Class: 
                </p>,
                <p>
                  Number of Passengers:
                </p>,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Model: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Model: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Class: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Class: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Number of Passengers:",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Number of Passengers:",
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Name: 
                </p>,
                <p>
                  Model: 
                </p>,
                <p>
                  Class: 
                </p>,
                <p>
                  Number of Passengers:
                </p>,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Model: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Model: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Class: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Class: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Number of Passengers:",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Number of Passengers:",
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Name: 
                </p>,
                <p>
                  Model: 
                </p>,
                <p>
                  Class: 
                </p>,
                <p>
                  Number of Passengers:
                </p>,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Model: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Model: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Class: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Class: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Number of Passengers:",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Number of Passengers:",
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Vehicles
          </h1>,
          <div
            className="button-container"
          >
            <button
              className="btn"
            >
              MAIN
            </button>
            <button
              className="btn"
            >
              PEOPLE
            </button>
            <button
              className="btn"
            >
              PLANETS
            </button>
          </div>,
          <div
            className="card-container"
          >
            <div
              className="card"
            >
              <p>
                Name: 
              </p>
              <p>
                Model: 
              </p>
              <p>
                Class: 
              </p>
              <p>
                Number of Passengers:
              </p>
            </div>
            <div
              className="card"
            >
              <p>
                Name: 
              </p>
              <p>
                Model: 
              </p>
              <p>
                Class: 
              </p>
              <p>
                Number of Passengers:
              </p>
            </div>
            <div
              className="card"
            >
              <p>
                Name: 
              </p>
              <p>
                Model: 
              </p>
              <p>
                Class: 
              </p>
              <p>
                Number of Passengers:
              </p>
            </div>
          </div>,
        ],
        "className": "Vehicles",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "title",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn"
              >
                MAIN
              </button>,
              <button
                className="btn"
              >
                PEOPLE
              </button>,
              <button
                className="btn"
              >
                PLANETS
              </button>,
            ],
            "className": "button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MAIN",
                "className": "btn",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "MAIN",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PEOPLE",
                "className": "btn",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "PEOPLE",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PLANETS",
                "className": "btn",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "PLANETS",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card"
              >
                <p>
                  Name: 
                </p>
                <p>
                  Model: 
                </p>
                <p>
                  Class: 
                </p>
                <p>
                  Number of Passengers:
                </p>
              </div>,
              <div
                className="card"
              >
                <p>
                  Name: 
                </p>
                <p>
                  Model: 
                </p>
                <p>
                  Class: 
                </p>
                <p>
                  Number of Passengers:
                </p>
              </div>,
              <div
                className="card"
              >
                <p>
                  Name: 
                </p>
                <p>
                  Model: 
                </p>
                <p>
                  Class: 
                </p>
                <p>
                  Number of Passengers:
                </p>
              </div>,
            ],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Name: 
                  </p>,
                  <p>
                    Model: 
                  </p>,
                  <p>
                    Class: 
                  </p>,
                  <p>
                    Number of Passengers:
                  </p>,
                ],
                "className": "card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Name: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Name: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Model: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Model: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Class: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Class: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Number of Passengers:",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Number of Passengers:",
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Name: 
                  </p>,
                  <p>
                    Model: 
                  </p>,
                  <p>
                    Class: 
                  </p>,
                  <p>
                    Number of Passengers:
                  </p>,
                ],
                "className": "card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Name: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Name: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Model: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Model: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Class: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Class: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Number of Passengers:",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Number of Passengers:",
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Name: 
                  </p>,
                  <p>
                    Model: 
                  </p>,
                  <p>
                    Class: 
                  </p>,
                  <p>
                    Number of Passengers:
                  </p>,
                ],
                "className": "card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Name: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Name: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Model: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Model: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Class: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Class: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Number of Passengers:",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Number of Passengers:",
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
